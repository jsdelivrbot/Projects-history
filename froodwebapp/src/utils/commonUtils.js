export const getRandomId = () => (
  Math.random().toString(36).substring(3, 7)
);

export const showPdfPreview = (pdfData) => {
  const file = new Blob([pdfData], { type: 'application/pdf' });
  const fileURL = URL.createObjectURL(file);
  window.open(fileURL);
};

export const checkNegative = value => (
  Number(value) < 0 ? 0 : Number(value)
);

export const toFixed2 = value => (
  value ? Number(value).toFixed(2) : '0.00'
);

export const getMaxId = (array) => {
  if (array.length) {
    return array.length && array.map(f => f.id).reduce((a, b) => Math.max(a, b)) + 1;
  }

  return 0;
};
