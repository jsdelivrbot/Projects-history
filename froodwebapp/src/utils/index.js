export * from './commonUtils';
export * from './inputParsers';
export * from './inputFormatters';
export * from './prepareHelpers';
export * from './renderHelpers';
export * from './sagaHelpers';
export * from './mapHelpers';
export * from './reduxHelpers/reduxHelpers';
export * from './reduxHelpers/mainContainer/mainContainerReduxHelpers';
export getMainContainerState from './reduxHelpers/mainContainer/getMainContainerState';
export getMainContainerReducer from './reduxHelpers/mainContainer/getMainContainerReducer';
export * from './staticData/operators';

