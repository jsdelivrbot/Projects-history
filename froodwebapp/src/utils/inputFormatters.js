/* eslint-disable import/prefer-default-export */
export const percentageFormatter = value => `${value}%`;

const usdFormatter = value => `$ ${value}`.replace(/\B(?=(\d{3})+(?!\d))/g, ',');

export const getFormatterByCurrency = (currency) => {
  switch (currency) {
    case 'USD':
      return usdFormatter;
    default:
      return usdFormatter;
  }
};
