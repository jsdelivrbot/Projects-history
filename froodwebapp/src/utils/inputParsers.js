/* eslint-disable import/prefer-default-export */

export const moneyParser = value => value.replace(/\$\s?|(,*)/g, '');

export const percentageParser = value => (value.replace('%', ''));
