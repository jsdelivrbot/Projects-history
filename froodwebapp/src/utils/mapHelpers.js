/* eslint-disable import/prefer-default-export */

/**
 * Used to get menu items for FormSelect
 * @param {Array} menuItems
 */
export const getMenuItems = menuItems => (
  menuItems.map(menuItem => ({ key: menuItem.id, value: menuItem.name }))
);

export const join2Arrays = (a, b) => [].concat(...a.map(aItem => b.map(bItem => [].concat(aItem, bItem))));

/**
 * Used to cross join values of array and optionally join values of the output arrays by symbol provided
 * @param {Array} a
 * @param {Array} b
 * @param {Array} c
 */
export const crossJoinArrays = (a, b, c) => {
  if (b && c && c.length > 0) {
    return crossJoinArrays(join2Arrays(a, b), c);
  } else if (b) {
    return join2Arrays(a, b);
  }
  return a;
};
