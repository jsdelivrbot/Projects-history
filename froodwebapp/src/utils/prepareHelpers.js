/**
 * Used for Main Container Filter Section
 * @param {array} filters
 */

export const prepareFilterValue = filters => (
  filters.map(filter => ({
    relationalOpId: filter.relationalOpId,
    columnId: filter.columnId,
    logicalOpId: filter.logicalOpId,
    value: filter.value
  }))
);

/**
 * Used in Main Container Table Section
 * @param {array} columns
 */
export const prepareHeaders = columns => (
  columns.map(col => ({
    id: col.id,
    key: col.dataAlias,
    name: col.name,
    order: col.order,
    dataAlias: col.dataAlias,
    dataType: col.tableDataType,
    isEditable: col.isEditable,
    isSortable: col.isSortable
  }))
);
