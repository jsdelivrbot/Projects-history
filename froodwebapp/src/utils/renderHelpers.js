/* eslint-disable import/prefer-default-export */
import React from 'react';
import { startCase } from 'lodash';

/**
 *
 * @param {array} params
 * @param {string} header
 */
export const renderAutocompleteItem = (params, header) => (item, isHighlighted) => (
  <div
    key={ item.id }
    style={ {
      backgroundColor: isHighlighted ? '#f7f7f7' : 'white',
      margin: '0.3rem',
      padding: '0.5rem 1rem',
      border: '3px solid #f7f7f7',
      borderRadius: '3px'
    } }
  >
    <span
      style={ {
        display: 'inline-block',
        verticalAlign: 'top',
        marginRight: '1rem'
      } }
    >
      { `${header} :` }
    </span>
    <span
      style={ {
        display: 'inline-block',
        textAlign: 'start'
      } }
    >
      <div>
        <span
          style={ {
            color: 'red',
            display: 'inline-block'
          } }
        >
          { startCase(`${params[0]}`) }
        </span>
        <span>{ ` - ${item[params[0]]}` }</span>
      </div>
      <div>
        <span
          style={ {
            color: 'blue',
            display: 'inline-block'
          } }
        >
          { startCase(`${params[1]}`) }
        </span>
        <span>{ ` - ${item[params[1]]}` }</span>
      </div>
      <div>
        <span
          style={ {
            color: 'green',
            display: 'inline-block'
          } }
        >
          { startCase(`${params[2]}`) }
        </span>
        <span>{ ` - ${item[params[2]]}` }</span>
      </div>
      { params[3] &&
        <div>
          <span
            style={ {
              color: 'brown',
              display: 'inline-block'
            } }
          >
            { startCase(`${params[3]}`) }
          </span>
          <span>{ ` - ${item[params[3]]}` }</span>
        </div>
      }
    </span>
  </div>
);
