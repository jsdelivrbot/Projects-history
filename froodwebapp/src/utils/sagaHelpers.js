/* eslint-disable import/prefer-default-export */

/**
 * Used in Sagas to automatically add params to urls
 * @param {object} filter
 * @param {string} url
 */
export const addParamsToURL = (filter, url) => {
  const params = [
    'orderNo',
    'id',
    'countryId',
    'locationId',
    'contactId',
    'mappingId',
    'slotId',
    'holidayId',
    'zoneId',
    'zoneType',
    'linkId',
    'addressId',
    'vendorId',
    'detailId',
    'itemId',
    'locationId',
    'slcId',
    'imageId'
  ];
  let newUrl = url;

  if (filter.limit) {
    newUrl += `?limit=${filter.limit}&offset=${filter.offset}`;
  }

  if (filter.limit && filter.sortBy) {
    newUrl += `&sortBy=${filter.sortBy}&sortOrder=${filter.sortOrder}`;
  }

  params.forEach((param) => {
    if (filter[param] && url.includes(param)) {
      newUrl = newUrl.replace(`{${param}}`, filter[param]);
    } else if (filter[param]) {
      newUrl += `&${param}=${filter[param]}`;
    }
  });

  // autocomplete logic
  newUrl = newUrl.includes('{keyword}') ? newUrl.replace('{keyword}', filter.payload) : newUrl;

  return newUrl;
};
