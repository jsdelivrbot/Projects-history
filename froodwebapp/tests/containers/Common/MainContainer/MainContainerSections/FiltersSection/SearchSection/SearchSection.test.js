import SearchSection from 'containers/Common/MainContainer/MainContainerSections/FiltersSection/SearchSection/SearchSection';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { prepareFilterValue } from 'utils';

const { describe, expect, it } = global;

describe('SearchSection', () => {
  const props = {
    activeFilter: {
      filterId: null,
      filterName: 'Name',
      value: [{
        logicalOpId: null,
        relationalOpId: 1,
        columnId: null,
        value: 'Some value',
      }],
      userDefined: false
    },
    filterColumns: [{
      id: 1,
      order: 'first order',
      displayName: 'Inventory',
      isDefault: false
    }, {
      id: 2,
      order: 'second order',
      displayName: 'Price lists',
      isDefault: true
    }],
    deleteFilterRequest: jest.fn(),
    saveFilterRequest: jest.fn(),
    updateFilterRequest: jest.fn(),
  };

  it('renders the SearchSection component', () => {
    const component = shallow(<SearchSection { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of SearchSection', () => {
    const component = shallow(<SearchSection { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
