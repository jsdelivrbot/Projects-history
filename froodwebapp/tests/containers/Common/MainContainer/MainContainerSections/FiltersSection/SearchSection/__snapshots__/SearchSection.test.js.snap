// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSection capturing Snapshot of SearchSection 1`] = `
<div
  className="searchSection"
>
  <Row>
    <Col
      lg={1}
      md={1}
      sm={1}
      xs={2}
    />
    <Col
      lg={2}
      md={3}
      sm={3}
      xs={3}
    >
      <Input
        disabled={false}
        onChange={[Function]}
        placeholder="Filter name.."
        prefixCls="ant-input"
        size="default"
        type="text"
      />
    </Col>
    <Col
      lg={true}
      md={true}
      sm={true}
      xs={true}
    >
      <Button
        className="saveFilterButton"
        onClick={[Function]}
        size="default"
      >
        Save
      </Button>
    </Col>
  </Row>
  <FilterRow
    filterColumns={
      Array [
        Object {
          "displayName": "Inventory",
          "id": 1,
          "isDefault": false,
          "order": "first order",
        },
        Object {
          "displayName": "Price lists",
          "id": 2,
          "isDefault": true,
          "order": "second order",
        },
      ]
    }
    handleAddFilterRow={[Function]}
    handleChangeColumn={[Function]}
    handleChangeColumnAvailableValue={[Function]}
    handleChangeLogicalOperator={[Function]}
    handleChangeRelationalOperator={[Function]}
    handleDatePickerChange={[Function]}
    handleDeleteFilterRow={[Function]}
    handleFilterInputChange={[Function]}
    key="0"
    rowData={
      Object {
        "columnId": null,
        "id": 0,
        "logicalOpId": null,
        "relationalOpId": 1,
        "value": "Some value",
      }
    }
    rowIndex={0}
  />
</div>
`;
